docs/ref/models/fields.txt

```diff
@@ -119,14 +122,91 @@ define a suitably-named constant for each value::
         )
 
         def is_upperclass(self):
-            return self.year_in_school in (self.JUNIOR, self.SENIOR)
+            return self.year_in_school in {self.JUNIOR, self.SENIOR}
 
 Though you can define a choices list outside of a model class and then
 refer to it, defining the choices and names for each choice inside the
 model class keeps all of that information with the class that uses it,
 and makes the choices easy to reference (e.g, ``Student.SOPHOMORE``
 will work anywhere that the ``Student`` model has been imported).
 
+In addition, Django provides special enumeration types you can subclass to
+define this iterable in a concise way::
+
+    from django.utils.translation import gettext_lazy as _
+
+    class YearInSchool(models.ChoiceStrEnum):
```

Comment:
So... do we want `YearInSchool` as a nested class inside `Student`, since we kind of recommend that usage?

Reply:
We can do. Sometimes people want to reuse the same set of choices for multiple models. Feel free to move it under `Student` if you think it makes the examples clearer.

Reply:
Ok. Have made the change.